### PR TITLE
Optimize: Use a positive number as the timeout of DNSEntry.

### PR DESCRIPTION
**Pull request by oknet**
`schedule_every` with a negative is more readable. so I change the
DNS_PERIOD to a postive value and call `schedule_every` with
`-DNS_PERIOD`.

It is a common design to reschedule the DNSEntry within
`MUTEX_RETRY_DELAY` cause fail on mutex lock.

### CODE REVIEW

**Comment by oknet** [comment]
```
dns/DNS.cc:1367:  e->timeout = h->mutex->thread_holding->schedule_in(e, DNS_PERIOD);
```

The `e` (DNSEntry) will be called back immediately because the DNS_PERIOD is a negative number.

**Comment by oknet** [comment]
[approve ci autest]

**Comment by oknet** [comment]
[approve ci autest]

**Comment by oknet** [comment]
@zwoop This is a candidate for 7.1.2.

**Comment by oknet** [comment]
[approve ci autest]

**Comment by zwoop** [comment]
[approve ci autest]

**Comment by zizhong** [comment]
@oknet I remember I tried to make the similar change and @SolidWallOfCode  talked me out of it. The discussion can be found https://github.com/apache/trafficserver/pull/1863.

**Comment by oknet** [comment]
@zizhong Thanks for your information.

/cc @SolidWallOfCode

In my opinion, `MUTEX_RETRY_DELAY` may be better than `DNS_PERIOD` for `schedule_in()`. 

@zwoop @bryancall  

`schedule_every` with a negative is more readable. so I change the DNS_PERIOD to a postive value and call `schedule_every` with `-DNS_PERIOD`.

**Comment by oknet** [comment]
[approve ci freebsd]

**Comment by SolidWallOfCode** [comment]
[approve ci freebsd]

**Comment by SolidWallOfCode** [comment]
[approve ci autest]

**Comment by oknet** [comment]
[approve ci autest]

**Comment by oknet** [comment]
[approve ci fedora]

**Comment by oknet** [comment]
[approved ci freebsd autest]

**Comment by oknet** [comment]
@zizhong @SolidWallOfCode 

It is a common design to reschedule the DNSEntry within `MUTEX_RETRY_DELAY` cause fail on mutex lock.

**Comment by oknet** [comment]
ping @zwoop

**Comment by zwoop** [comment]
Oops, missed the ping here. I'm testing this on Docs right now, should we land this for 7.1.2 ?

**Comment by oknet** [comment]
@zwoop yes, it is still a candidate for 7.1.2.

File: iocore/dns/DNS.cc [hunk 1]
```diff
@@ -1364,7 +1364,7 @@ dns_result(DNSHandler *h, DNSEntry *e, HostEnt *ent, bool retry, bool tcp_retry)
   if (e->timeout) {
     e->timeout->cancel();
   }
-  e->timeout = h->mutex->thread_holding->schedule_in(e, DNS_PERIOD);
```
Comment (oknet) [inline iocore/dns/DNS.cc]: For retry later, it should be `DNS_DELAY_PERIOD` or `MUTEX_RETRY_DELAY`.

**Comment by zwoop** [comment]
Cherry-picked to 7.1.2. It had a tiny merge conflict, I'm pretty confident I got it right though but please double check the 7.1.x branch.

**Comment by oknet** [comment]
@zwoop The commit 65e544688b8b0e677ce1606948cc247a224e3a05 is correct.

**Pull request merged**